templates/custom/projects-cm.yaml

```diff
@@ -0,0 +1,91 @@
+{{- $root := . -}}
+{{- range $project := .Values.projects.project }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $project.name}}-project-cm
+data:
+  projects.sh: |
+   #!/usr/bin/env bash
+
+    # exit when any command fails
+    set -e
+
+    coreHostUrl={{ $root.Values.expose.ingress.hosts.core }}
+    basicAuthCreds="${HARBOR_ADMIN_USERNAME}:${HARBOR_ADMIN_PASSWORD}"
+
+    function isProjectExists(){
+      # call project API and return the value.
+      res=$(curl -I HEAD "$coreHostUrl/api/projects?project_name=$1" \
+      -H "accept: application/json" \
+      -u $basicAuthCreds)
+      echo "response: " $res
+    }
+
+    function createProject() {
+      # call Project API and create the project with the values taken from the values yaml
+      project=$(curl -X POST "$coreHostUrl/api/projects" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_ADMIN_USERNAME}":"${HARBOR_PASSWORD}" \
```

Comment:
replace with `basicAuthCreds`

Reply:
Resolved and use variable 'basicAuthCreds'